packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts

```diff
@@ -425,6 +432,64 @@ export class StripeWebhookHandler extends StripeHandler {
     await this.updateCustomer(uid, email);
   }
 
+  /**
+   * Validate plan metadata on update
+   */
+  async handlePlanUpdatedEvent(request: AuthRequest, event: Stripe.Event) {
+    const plan = event.data.object as Stripe.Plan;
+    if (typeof plan.product !== 'string') return;
```

Comment:
Does this actually happen?  Stripe should be pretty clear about what they are passing in the webhook event.

Reply:
![image](https://user-images.githubusercontent.com/1844554/121108842-741f3e00-c7d8-11eb-8508-c58f51b81ccf.png)

Reply:
Sure, I understand what Stripe declared the type of Plan.product to be.  But in https://stripe.com/docs/api/events/types, it states that object for the event is a https://stripe.com/docs/api/plans/object, which has a expandable `product`.  afaik, the expandable properties are not automatically expanded.

If we were to handle the types accordingly here, then we should not return when `product` is a Product.  We should instead use the product, since that's the object we want on line 441.

Reply:
If we do expect Stripe to return any of the types in the sum type for `Stripe.Plan.product`, then we need to handle the case when it is a `Stripe.Product`.  That's a valid value and returning early would be a bug.

However, I do not think that value is ever a `Stripe.Product` or `Stripe.DeletedProduct`, since that property does not auto-expand.

If you are using that type check to get around the type checker, then use a type assertion instead.  This is one of the times where we know more than the type checker.